### ISSUE TITLE
Button Description Accessibility

### ISSUE THREAD

**Issue by irmerk** [accordproject/cicero-ui#55]
# WCAG 2.0 Level A [2.4.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html) (Link Purpose)

## Tasks
- Ambiguous buttons should be more descriptive for screen readers. This can be achieved by creating a visually hidden span with a description in it.

## Related Issues: 
- Issue [#40](https://github.com/accordproject/template-studio-v2/issues/40) in Template Studio v2

**Comment by irmerk** [comment]
### Hacktoberfest:
This issue can possibly provide for multiple pull requests.

**Comment by Anisha1234** [comment]
Hello @irmerk . I am interested in this issue. I installed the node_moules and tried running the project with npm run start. The code is getting compiled successfully but I am getting a completely blank page in my localhost. The error I am receiving is, ```Uncaught TypeError: Cannot read property 'div' of undefined```. Please help me resolve the issue.

**Comment by irmerk** [comment]
@Anisha1234 for reference, this is being solved here: https://github.com/accordproject/markdown-editor/issues/90

**Comment by Anisha1234** [comment]
@irmerk I am being solve the setup issue by referencing #147. So from now, I would love to focus on the current issue.  I would appriciate more info on the issue. In my localhost, I can see the following with `waiting`
![Screenshot_20191001_194903](https://user-images.githubusercontent.com/16955978/65970704-a4033380-e484-11e9-8a5c-42ceaf01eb8f.png)

**Comment by irmerk** [comment]
@Anisha1234 You'll most likely want to develop `cicero-ui` within a web-app for the time being. Right now that's easiest in `template-studio-v2`, so you'll use [these instructions](https://github.com/accordproject/cicero-ui#development) and use `template-studio-v2` for your "_main app directory_".

**Comment by Anisha1234** [comment]
@irmerk yes, I am being able to run the template-studio-v2 project as well. I would like to know which Ambiguous buttons are we referring here.

**Comment by irmerk** [comment]
I originally opened this as an open and general issue, mostly following these two links:

1. https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=244#link-purpose-in-context
2. https://www.wuhcag.com/link-purpose-in-context/

I think this will most likely require using a screen reader to go through trial and error to see how links and buttons respond to that software.

**Comment by Anisha1234** [comment]
@irmerk I used ChromeVox for trial and error and the link descriptions seem to work fine. Do you want me to give more detailed description to them?

**Comment by irmerk** [comment]
> @irmerk I used ChromeVox for trial and error and the link descriptions seem to work fine. Do you want me to give more detailed description to them?

This could very well be the case. This could also give some indication that more buttons need to be accessible, and thus have descriptions. For instance, the `+ Add to contract` and `Details` buttons in the `TemplateLibrary` on the right of `template-studio-v2` aren't accessible via they keyboard. So we're not seeing that they could very well not have accessible descriptions, either.

These kind of branching off issues could be made into new issues or worked on from this issue.

**Comment by sanyamdogra** [comment]
Hi @irmerk,  a `<span>` tag with a `className='sr-only'` can be used to add textual data only for screen readers. I am new to this project, please correct me if wrong. Thanks!

**Comment by irmerk** [comment]
@sanyamdogra I'm not as familiar with `sr-only`, but it looks like that's in [Bootstrap](https://v4-alpha.getbootstrap.com/getting-started/accessibility/) for a [Skip To Main](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html) functionality for screen readers.

**Comment by adrijshikhar** [comment]
@irmerk i would like to work on this issue, can you please assign to me. Thanks

**Comment by sanyamdogra** [comment]
Hi @irmerk, why can't we use `aria-label` property to add a description for a button?

**Comment by irmerk** [comment]
> Hi @irmerk, why can't we use `aria-label` property to add a description for a button?

Unsure why we can't?

**Comment by sanyamdogra** [comment]
I went through semantic UI docs and found that adding an `aria-label` to the Buttons will lead to a HTML code with the same `aria-label` for screen readers.
![image](https://user-images.githubusercontent.com/33497630/76289778-dfc1e500-62cf-11ea-8e23-2b57f34bd05f.png)
Ref:-https://react.semantic-ui.com/elements/button/#types-button

I can start by adding these from component to component? We can create a separate issue with list of components utilizing Buttons, maybe to create a track.  What's your take on this? @irmerk 
Thanks

**Comment by irmerk** [comment]
I'm not sure I follow. `semantic-ui-react` supports aria-label, and you're saying we can't use it? But also that we should add these?

**Comment by sanyamdogra** [comment]
Nope I didn't meant that we can't use them. I am saying that we should use them to enhance button description accessibility, thus fixing this issue. My question is that, will adding `aria-label` solve this issue? Or you are expecting something else.

**Comment by irmerk** [comment]
Ah okay. Yes, that is what I had in mind.

**Comment by sanyamdogra** [comment]
Alright, I'll get on with the PR then. I guess there are 2 places that have buttons:
* TemplateLibrary > Buttons
* Demo

Am I missing some?

**Comment by irmerk** [comment]
I believe so.
